ecgtools/core.py

```diff
@@ -213,21 +216,125 @@ def parse_files_attributes(
         filepaths = list(itertools.chain(*filelist))
         self.entries = parse_files_attributes(filepaths, global_attrs, parser, lazy, nbatches)
         return self
 
-    def to_df(self):
+    def create_catalog(
+        self,
+        catalog_file: str,
+        path_column: str,
+        variable_column: str,
+        data_format: str = None,
+        format_column: str = None,
+        groupby_attrs: list = None,
+        aggregations: List[dict] = None,
+        esmcat_version: str = None,
+        cat_id: str = None,
+        description: str = None,
+        attributes: List[dict] = None,
+        **kwargs,
+    ):
+
         """
-        Create catalog as a Pandas DataFrame.
+        Create a catalog, write it to a comma-separated
+        values (CSV) file, and create a corresponding JSON file
+        according to ESM collection specification defined at
+        https://github.com/NCAR/esm-collection-spec/.
+
+        Parameters
+        ----------
+        catalog_file : str
+           Path to a the CSV file in which catalog contents will be persisted.
+        path_column : str
+           The name of the column containing the path to the asset.
+           Must be in the header of the CSV file.
+        variable_column : str
+            Name of the attribute column in csv file that contains the variable name.
+        data_format : str
+            The data format. Valid values are netcdf and zarr.
+            If specified, it means that all data in the catalog is the same type.
+            Mutually exclusive with `format_column`.
+        format_column : str
+            The column name which contains the data format, allowing for multiple data assets
+            (file formats) types in one catalog. Mutually exclusive with `data_format`.
+        groupby_attrs : list
+            Column names (attributes) that define data sets that can be aggegrated,
+            default None
+        aggregations : List[dict]
+            List of aggregations to apply to query results, default None
+        esmcat_version : str
+            The ESM Catalog version the collection implements, default None
+        cat_id : str
+            Identifier for the collection, default None
+        description : str
+            Detailed multi-line description to fully explain the collection,
+            default None
+        attributes : List[dict]
+            A list of attributes. An attribute dictionary describes a column
+            in the catalog CSV file.
+        kwargs : Additional keyword arguments are passed through to the
+                 :py:class:`~pandas.DataFrame.to_csv` method.
 
         Returns
         -------
         `ecgtools.Builder`
 
+        Notes
+        -----
+        See https://github.com/NCAR/esm-collection-spec/blob/master/collection-spec/collection-spec.md
+        for more details on ESM collection specification.
+
         """
+
         self.df = pd.DataFrame(self.entries)
```

Comment:
If you want this to be publicly available, then I recommend moving this to the `Builder.parse_files_attributes` function (and renaming the function).  And make `self.entries` private.